solr/core/src/java/org/apache/solr/cloud/Overseer.java

```diff
@@ -219,6 +221,8 @@ else if (LeaderStatus.YES != isLeader) {
                 @Override
                 public void onEnqueue() throws Exception {
                   workQueue.offer(data);
+                  stateUpdateQueue.poll();
```

Comment:
I know right?  I honestly don't actually understand why there are two queues at all, I feel like it could be rewritten in terms of a single queue.